src/EnergyPlus/General.cc

```diff
@@ -333,10 +513,23 @@ namespace General {
 			DY = Y0 - Y1;
 			if ( std::abs( DY ) < SMALL ) DY = SMALL;
 			// new estimation
-			XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			if ( ( HVACSystemRootFinding.TypeNum == Bisection ) ||
+				( HVACSystemRootFinding.TypeNum == Alternation && AltIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == RegulaFalsiThenBisection && NIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == BisectionThenRegulaFalsi && NIte <= HVACSystemRootFinding.NumOfIter ) ) {
+				if ( HVACSystemRootFinding.TypeNum == Alternation ) {
+					if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
+				}
+				// Bisection
+				XTemp = ( X1 + X0 ) / 2.0;
+			} else {
+				// Regula Falsi
+				XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			}
```

Comment:
@rraustad Here is a section extracted from MSDN:

// switch_statement1.cpp  
#include <stdio.h>  
  
int main() {  
   char *buffer = "Any character stream";  
   int capa, lettera, nota;  
   char c;  
   capa = lettera = nota = 0;  
  
   while ( c = *buffer++ )   // Walks buffer until NULL  
   {  
      switch ( c )  
      {  
         case 'A':  
            capa++;  
            break;  
         case 'a':  
            lettera++;  
            break;  
         default:  
            nota++;  
      }  
   }  
   printf_s( "\nUppercase a: %d\nLowercase a: %d\nTotal: %d\n",  
      capa, lettera, (capa + lettera + nota) );  
}  

Break is necessary unless with case Default.